### PR TITLE
Revamped folder

### PR DESCRIPTION
**Pull request by maxhelias**
I would like to suggest a reorganization of the folder.
This allows to follow the recommendations of composer about the autoload-dev : https://getcomposer.org/doc/04-schema.md#autoload-dev and simplifies the integration of dev tools

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /.idea/
+/.php_cs.cache
```
Comment (odolbeau) [inline .gitignore]: Should be put in your global `.gitignore` ideally

File: src/Serializer/Normalizer/PhoneNumberNormalizer.php [hunk 2]
```diff
@@ -85,6 +85,10 @@ public function supportsNormalization($data, $format = null)
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
+        if (null === $data) {
```
Comment (odolbeau) [inline src/Serializer/Normalizer/PhoneNumberNormalizer.php]: Should be in another PR I think?

File: tests/Serializer/Normalizer/PhoneNumberNormalizerTest.php [hunk 3]
```diff
@@ -72,6 +72,7 @@ public function testDenormalize()
         $normalizer = new PhoneNumberNormalizer($phoneNumberUtil->reveal());
 
         $this->assertSame($phoneNumber, $normalizer->denormalize('+33193166989', 'libphonenumber\PhoneNumber'));
+        $this->assertSame(null, $normalizer->denormalize(null, 'libphonenumber\PhoneNumber'));
```
Comment (odolbeau) [inline tests/Serializer/Normalizer/PhoneNumberNormalizerTest.php]: Same (another PR)

File: src/Serializer/Normalizer/PhoneNumberNormalizer.php (hunk 2 above)
Comment (maxhelias) [reply src/Serializer/Normalizer/PhoneNumberNormalizer.php]: You're right, it's a test I forgot to remove

File: .gitignore (hunk 1 above)
Comment (OskarStark) [reply .gitignore]: I would prefer to keep it in the project, as it is generated IN the repo and should not accidently be commited!

Comment (maxhelias) [reply .gitignore]: I agree with @OskarStark, i give it back

Comment (Taluu) [reply .gitignore]: Yes and no ; yes, it is generated in the project... but no, because it's the result of a *tool* used on the project.

Otherwise, we'd need to do the same with the phpstorm config, ... and so on. And we'd never get out.

File: composer.json [hunk 4]
```diff
@@ -36,11 +36,13 @@
...
-        },
-        "exclude-from-classmap": [
-            "/Tests/"
-        ]
+            "Misd\\PhoneNumberBundle\\": "src/"
+        }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "Misd\\PhoneNumberBundle\\Tests\\": "tests/"
```
Comment (Taluu) [inline composer.json]: suffixing tests with a `Test` namespace is a "meh" practice at best. It's better imo to have the tests in the same namespace.

File: .gitignore (hunk 1 above)
Comment (maxhelias) [reply .gitignore]: @Taluu I do not follow you, we already ignore the phpstorm configuration, it is an external tool but that modify the project

File: composer.json (hunk 4 above)
Comment (maxhelias) [reply composer.json]: I'am not agree for this point, as the composer doc says :
`Classes needed to run the test suite should not be included in the main autoload rules to avoid polluting the autoloader in production and when other people use your package as a dependency.
`

File: .gitignore (hunk 1 above)
Comment (jderusse) [reply .gitignore]: agreed with @maxhelias : given.we provide a `.php_cs.dist` file, this is related to the bundle. like `vendor` is related to project even if composer is an external tool.

But `.idea` is definitively related to the developer and should be removed

File: composer.json (hunk 4 above)
Comment (Taluu) [reply composer.json]: There's s light difference here. It does not say "not to use the same namespace". Just "not the same autoloading rules", which is fine then in the `autoload-dev` section.

Comment (maxhelias) [reply composer.json]: Yes but we need to differ them

Comment (Taluu) [reply composer.json]: No. 

```json
{
	"autoload": {
		"psr-4": {
			"App\\": "src"
		
		}
	},

	"autoload-dev": {
		"psr-4": {
			"App\\": "tests"
		}
	}
}
```

works wonder. When doing a `composer install` from the root, then `App\\` prefix will load from both `src` and `tests`. When using `--no-dev`, it will only look in `src`, the same when this package isn't the root package.

Comment (maxhelias) [reply composer.json]: Good to know, thanks :)
WDYT @odolbeau ?

Comment (odolbeau) [reply composer.json]: I'm used to suffix the test namespace with `Test` also but TBH, I don't know why! :D
I don't have any opinion on this, both are fine.

It's not linked to this PR anyway. :)
